### PR TITLE
Initial commit for blackjack v1.

### PR DESCRIPTION
**Pull request by quinnvaughn**
## Summary
PR to show how to build a decentralized blackjack game. Version 1 that JP asked me to create.

### CODE REVIEW

File: examples/blackjack/index.rsh [hunk 1]
```diff
@@ -0,0 +1,171 @@
+"reach 0.1"
+
+const [isOutcome, ALICE_WINS, DRAW, BOB_WINS] = makeEnum(3)
+
+const CardSuit = UInt
+const CardValue = UInt
+
+const Card = Object({
+  suit: CardSuit,
```
Comment (jeapostrophe) [inline examples/blackjack/index.rsh:9]: You never look at the suit... it doesn't matter

File: examples/blackjack/index.rsh [hunk 2]
```diff
@@ -0,0 +1,171 @@
+"reach 0.1"
+
+const [isOutcome, ALICE_WINS, DRAW, BOB_WINS] = makeEnum(3)
+
+const CardSuit = UInt
+const CardValue = UInt
+
+const Card = Object({
```
Comment (jeapostrophe) [inline examples/blackjack/index.rsh:8]: So, card should just be the value

File: examples/blackjack/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Decentralized Blackjack on Reach
+This projects shows how to have a decentralized version of blackjack on the blockchain.
```
Comment (TheChronicMonster) [inline examples/blackjack/README.md:2]: This project

File: examples/blackjack/index.rsh [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    !hasBlackjackA && !hasBlackjackB,
+    getHandValue(initialHandA),
+    getHandValue(initialHandB),
+  ])
+    .invariant(balance() == 2 * wager)
+    .while(keepGoing)
+    .case(
+      Alice,
+      () => ({
+        when: declassify(interact.keepGoing(handValA)),
```
Comment (jeapostrophe) [inline examples/blackjack/index.rsh:127]: I don't think that this program should have `keepGoing` and instead the user should be forced to say "I don't want anything" or "I do want something"

File: examples/blackjack/index.rsh [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+      (_) => {
+        commit()
+        Bob.only(() => {
+          const card = declassify(interact.hit(handValB))
+        })
+        Bob.publish(card)
+        return [true, handValA, getNextHandValue(card, handValB)]
+      }
+    )
+    .timeout(relativeTime(DEADLINE), () => {
```
Comment (jeapostrophe) [inline examples/blackjack/index.rsh:152]: And then this won't be needed

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

**Comment by jeapostrophe** [comment]
Closing this because it is such a bigger problem than this.

**Pull request closed without merging**